src/core/lib/iomgr/socket_windows.cc

```diff
@@ -181,4 +181,24 @@ int grpc_ipv6_loopback_available(void) {
   return g_ipv6_loopback_available;
 }
 
+SOCKET grpc_create_wsa_socket(int family, int type, int protocol,
+                              LPWSAPROTOCOL_INFO protocol_info, GROUP group,
+                              DWORD flags) {
+  bool is_wsa_no_handle_inherit_set = flags & WSA_FLAG_NO_HANDLE_INHERIT;
+  if (!g_is_wsa_no_handle_inherit_supported && is_wsa_no_handle_inherit_set) {
+    flags ^= WSA_FLAG_NO_HANDLE_INHERIT;
+  }
+  SOCKET sock = WSASocket(family, type, protocol, protocol_info, group, flags);
+  /* WSA_FLAG_NO_HANDLE_INHERIT may be not supported on the older Windows
+     versions, see
+     https://msdn.microsoft.com/en-us/library/windows/desktop/ms742212(v=vs.85).aspx
+     for details. */
+  if (sock == INVALID_SOCKET && is_wsa_no_handle_inherit_set) {
+    g_is_wsa_no_handle_inherit_supported = false;
```

Comment:
this is not threadsafe, we need a better way of detecting if WSA_FLAG_NO_HANDLE_INHERIT is supported.

Reply:
Can you please suggest the best place to check that the host supports this flag? grpc_iomgr_init()?